packages/server/src/fhir/lookups/reference.ts

```diff
@@ -33,6 +34,28 @@ export class ReferenceTable extends LookupTable {
     const values = getSearchReferences(resource);
     await this.insertValuesForResource(client, resource.resourceType, values);
   }
+
+  /**
+   * Inserts reference values into the lookup table for a resource.
+   * @param client - The database client.
+   * @param resourceType - The resource type.
+   * @param values - The values to insert.
+   */
+  async insertValuesForResource(
+    client: Pool | PoolClient,
+    resourceType: ResourceType,
+    values: Record<string, any>[]
+  ): Promise<void> {
+    if (values.length === 0) {
+      return;
+    }
+    const tableName = this.getTableName(resourceType);
+
+    // Reference lookup tables have a covering primary key, so a conflict means
+    // that the exact desired row already exists in the database
+    const insert = new InsertQuery(tableName, values).ignoreOnConflict();
```

Comment:
Nice, very clean

Do you mind adding a test to check that `pg` doesn't actually emit an error in this case? (I think that's what was happening previously right?)

Reply:
The conflict is relatively difficult to reproduce reliably in a test, since it's effectively a race condition in the database